### PR TITLE
fix(sbb-navigation): focus outline on the trigger element in Safari

### LINKED ISSUE #1776: bug(sbb-navigation): focus outline wrongly displayed in Safari
Safari: when closing navigation via mouse the keyboard focus is still visible on the navigation trigger (Hint: See tooltip solution)

Problem in Action (only Safari): https://digital-next.app.sbb.ch/de/foundation/assets/icons/

### PR DESCRIPTION
**Pull request by dauriamarco**
Closes #1776.

### CODE REVIEW

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Thanks! I think we need to fix this for every overlay component (using native dialog) and I‘d appreciate a global function for it

File: src/components/sbb-navigation/sbb-navigation.tsx [hunk 1]
```diff
@@ -233,8 +234,28 @@ export class SbbNavigation implements ComponentInterface {
...
+              () => delete this._triggerElement.dataset.focusOrigin,
+              { once: true }
+            );
+          },
+          { once: true }
+        );
+      }
+
       this._navigation.close();
+      // To enable focusing other element than the trigger, we need to call focus() a second time.
```
Comment (jeripeierSBB) [inline src/components/sbb-navigation/sbb-navigation.tsx:257]: The comment is not reallly valid as we focus the trigger in this case

File: src/global/helpers/focus.ts [hunk 2]
```diff
@@ -29,6 +31,33 @@ export function getFocusableElements(
...
 
+export function focusAndSetCloseModality(
+  elementToFocus: HTMLElement,
+  focusOptions?: FocusOptions
+): void {
+  if (!elementToFocus) {
+    return;
+  }
+
+  const closedByModality = sbbInputModalityDetector.mostRecentModality;
```
Comment (jeripeierSBB) [inline src/global/helpers/focus.ts:42]:
```suggestion
  const lastModality = sbbInputModalityDetector.mostRecentModality;
```

File: src/global/helpers/focus.ts [hunk 3]
```diff
@@ -29,6 +31,33 @@ export function getFocusableElements(
   return [...focusableEls];
 }
 
+export function focusAndSetCloseModality(
```
Comment (jeripeierSBB) [inline src/global/helpers/focus.ts:34]:
```suggestion
export function focusAndSetModality(
```

Comment (jeripeierSBB) [inline src/global/helpers/focus.ts:34]: Maybe add a short description

File: src/components/sbb-tooltip/sbb-tooltip.tsx [hunk 4]
```diff
@@ -399,23 +402,7 @@ export class SbbTooltip implements ComponentInterface {
...
-              { once: true }
-            );
-          },
-          { once: true }
-        );
-      }
+      focusAndSetCloseModality(elementToFocus);
       this._dialog.close();
       // To enable focusing other element than the trigger, we need to call focus() a second time.
       elementToFocus?.focus();
```
Comment (jeripeierSBB) [inline src/components/sbb-tooltip/sbb-tooltip.tsx:408]:
```suggestion
```

File: src/components/sbb-tooltip/sbb-tooltip.tsx [hunk 5]
```diff
@@ -399,23 +402,7 @@ export class SbbTooltip implements ComponentInterface {
...
-            elementToFocus.addEventListener(
-              'blur',
-              () => delete elementToFocus.dataset.focusOrigin,
-              { once: true }
-            );
-          },
-          { once: true }
-        );
-      }
+      focusAndSetCloseModality(elementToFocus);
```
Comment (jeripeierSBB) [inline src/components/sbb-tooltip/sbb-tooltip.tsx:405]: I think we should move this after close()

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Thanks, almost done :P

File: src/components/sbb-tooltip/sbb-tooltip.tsx (hunk 5 above)
Comment (dauriamarco) [reply src/components/sbb-tooltip/sbb-tooltip.tsx]: If I move it after close I noticed that it doesn't work anymore...

File: src/components/sbb-dialog/sbb-dialog.tsx [hunk 6]
```diff
@@ -290,6 +290,7 @@ export class SbbDialog implements ComponentInterface {
     } else if (event.animationName === 'close') {
       this._state = 'closed';
       this._dialogWrapperElement.querySelector('.sbb-dialog__content').scrollTo(0, 0);
+      seModalityOnNextFocus(this._lastFocusedElement);
```
Comment (jeripeierSBB) [inline src/components/sbb-dialog/sbb-dialog.tsx:293]:
```suggestion
      setModalityOnNextFocus(this._lastFocusedElement);
```

File: src/components/sbb-dialog/sbb-dialog.tsx [hunk 7]
```diff
@@ -290,6 +290,7 @@ export class SbbDialog implements ComponentInterface {
...
       // Manually focus last focused element in order to avoid showing outline in Safari
       this._lastFocusedElement?.focus();
```
Comment (jeripeierSBB) [inline src/components/sbb-dialog/sbb-dialog.tsx:295]: should we call it after close to be consistent with tooltip and navigation?

File: src/components/sbb-tooltip/sbb-tooltip.tsx [hunk 8]
```diff
@@ -348,7 +347,12 @@ export class SbbTooltip implements ComponentInterface {
   // Close tooltip on backdrop click.
   private _closeOnBackdropClick = async (event: PointerEvent): Promise<void> => {
     if (!this._isPointerDownEventOnTooltip && !isEventOnElement(this._dialog, event)) {
-      this._nextFocusedElement = document.activeElement as HTMLElement;
+      const composedPathElements = event
```
Comment (jeripeierSBB) [inline src/components/sbb-tooltip/sbb-tooltip.tsx:350]: should we also use our method getEventTarget here?

File: src/global/helpers/focus.ts [hunk 9]
```diff
@@ -29,6 +31,30 @@ export function getFocusableElements(
   return [...focusableEls];
 }
 
+// Set the input modality in order to avoid showing the outline in Safari.
+export function seModalityOnNextFocus(elementToFocus: HTMLElement): void {
```
Comment (jeripeierSBB) [inline src/global/helpers/focus.ts:35]: maybe this method would better fit to the input-modality file?

Comment (dauriamarco) [reply src/global/helpers/focus.ts:35]: Ah yes, I forgot to ask the same question. I'm not sure though, would you move it there? And inside or outside the class?

Comment (dauriamarco) [reply src/global/helpers/focus.ts:35]: Agree

File: src/components/sbb-tooltip/sbb-tooltip.tsx [hunk 10]
```diff
@@ -348,7 +348,12 @@ export class SbbTooltip implements ComponentInterface {
...
+        .composedPath()
+        .filter((el) => el instanceof window.HTMLElement);
+      this._nextFocusedElement = composedPathElements.find((el) =>
+        (el as HTMLElement).matches(IS_FOCUSABLE_QUERY)
+      ) as HTMLElement;
```
Comment (jeripeierSBB) [inline src/components/sbb-tooltip/sbb-tooltip.tsx:356]:
```suggestion
      this._nextFocusedElemen = event
        .composedPath()
        .filter((el) => el instanceof window.HTMLElement)
        .find((el) =>
        (el as HTMLElement).matches(IS_FOCUSABLE_QUERY)
      ) as HTMLElement;
```

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Soon we have it :D

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Soon we have it :D

**Review by jeripeierSBB — APPROVED** [review]
LGTM, thanks!

**Pull request merged**